nixos/modules/services/networking/pounce.nix

```diff
@@ -0,0 +1,195 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pounce;
+  defaultUser = "pounce";
+  configFile = pkgs.writeText "pounce" ''
+    ${optionalString (cfg.sslCa != null) "local-ca = ${cfg.sslCa}"}
+    local-host = ${cfg.hostAddress}
+    local-port = ${toString cfg.hostPort}
+    local-priv = ${cfg.sslKey}
+    local-cert = ${cfg.sslCert}
+    join = ${concatStringsSep "," cfg.joinChannels}
+    ${optionalString (cfg.hashedPass != "") "local-pass = "+cfg.hashedPass}
+    host = ${cfg.ircHost}
+    port = ${toString cfg.ircHostPort}
+    user = ${cfg.ircUser}
+    nick = ${cfg.ircNick}
+    ${optionalString (cfg.certFP != null) "client-cert = "+cfg.certFP+"\nsasl-external"}
+    ${optionalString (cfg.ircPass != "") "pass = "+cfg.ircPass}
+    ${cfg.extraConfig}
+  '';
+in
+{
+  options = {
+    services.pounce = {
+      enable = mkEnableOption "the Pounce IRC bouncer";
+
+      user = mkOption {
+        default = defaultUser;
+        example = "john";
+        type = types.str;
+        description = ''
+          The name of an existing user account to use to own the Pounce server
+          process. If not specified, a default user will be created.
+        '';
+      };
+
+      group = mkOption {
+        default = defaultUser;
+        example = "users";
+        type = types.str;
+        description = ''
+          Group to own the Pounce process.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.pounce;
+        defaultText = literalExpression "pkgs.pounce";
+        description = ''
+          Set version of pounce package to use.
+        '';
+      };
+
+      hostAddress = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          Where pounce should listen for incoming connections.
+        '';
+      };
+
+      hostPort = mkOption {
+        type = types.port;
+        default = 6697;
+        description = ''
+          The port on which Pounce listens.
+        '';
+      };
+
+      sslCa = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "~/.config/pounce/pouncecert.pem";
```

Comment:
I don't think this example will work (types.path checks for / at start of string)